lib/ansible/modules/cloud/openstack/os_subnet.py

```diff
@@ -310,7 +318,8 @@ def main():
                     host_routes=host_routes,
                     ipv6_ra_mode=ipv6_ra_mode,
                     ipv6_address_mode=ipv6_a_mode,
-                    tenant_id=project_id)
+                    tenant_id=project_id,
+                    **extra_specs)
```

Comment:
this is error-prone as it will cause exception if someone decides to have duplicate keys there. better way is to do like this:
```python
kwargs = dict(...)  # leave it as is
if any(spec in kwargs for spec in extra_specs):  # do input validation
    raise ValueError('Duplicate key in extra_specs')
kwargs = dict(kwargs, **extra_specs)  # actually upgrade it
```